### PR TITLE
Logilink EC0003 functionality

### PR DESCRIPTION
**Pull request by hernae**
I copied the original logilink_switch and made the needed changes.

I didn't compile the full project (for testing), just added the new protocol as a module with:
 gcc -fPIC -shared libs/pilight/protocols/433.92/logilink_switch2.c -Ilibs/pilight/protocols/ -Iinc/ -o logilink_switch2.so -DMODULE=1

See forum:
https://forum.pilight.org/Thread-Logilink-EC0003-Protocol

### CODE REVIEW

File: libs/pilight/protocols/433.92/logilink_switch2.c [hunk 1]
```diff
@@ -0,0 +1,234 @@
+/*
+	Copyright (C) 2015 CurlyMo, Meloen and Scarala
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/logilink_switch2.c]: Can you fix this license block and also replace it with the MPLv2?

File: CMakeConfig.txt [hunk 2]
```diff
@@ -56,6 +56,7 @@ set(PROTOCOL_LM76 ON CACHE BOOL "support for the lm76 i2c temperature sensor")
 set(PROTOCOL_LIRC ON CACHE BOOL "support for the Lirc API")
 set(PROTOCOL_LIVOLO_SWITCH ON CACHE BOOL "support for the Livolo switch protocol")
 set(PROTOCOL_LOGILINK_SWITCH ON CACHE BOOL "support for the Logilink switch protocol")
+set(PROTOCOL_LOGILINK_SWITCH2 ON CACHE BOOL "support for the Logilink EC0003 switch protocol")
```
Comment (CurlyMoo) [inline CMakeConfig.txt]: Can you check what type was supported by the other logilink protocol?

**Comment by CurlyMoo** [comment]
1. Can you fix your commit so it's properly linked to your account?
2. I also miss `CMakeExclude.txt` entry.
3. Can you make a PR for the rewrite branch as well including the unit test?

File: libs/pilight/protocols/433.92/logilink_switch2.c [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+		}
+
+		x -= 2;
+	}
+}
+
+static void createUnitCode(int unitcode) {
+	switch(unitcode) {
+		case 7:
+			createHigh(40, 45);	// Button 1
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/logilink_switch2.c]: Can you check the alignment here?

File: libs/pilight/protocols/433.92/logilink_switch2.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/*
+	Copyright (C) 2015 CurlyMo, Meloen and Scarala
+
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/logilink_switch2.h]: Can you change this license to the MPLv2 as well?

File: CMakeConfig.txt (hunk 2 above)
Comment (hernae) [reply CMakeConfig.txt]: Logilink EC0002, EC0004, EC0005 and EC0006

File: libs/pilight/protocols/433.92/logilink_switch2.c (hunk 3 above)
Comment (hernae) [reply libs/pilight/protocols/433.92/logilink_switch2.c]: Did you mean the space behind createHigh(40, 45);?

Comment (hernae) [reply libs/pilight/protocols/433.92/logilink_switch2.c]: 1. Can you fix your commit so it's properly linked to your account?
-> What do you mean with fixing the commit and linking it to my account? 
2. I also miss CMakeExclude.txt entry.
-> done
3. Can you make a PR for the rewrite branch as well including the unit test?
-> Where do I find the unit tests?

File: CMakeConfig.txt (hunk 2 above)
Comment (CurlyMoo) [reply CMakeConfig.txt]: Maybe we should add these version numbers to the other protocol as well:
`support for the Logilink switch protocol EC0002,4,5,6`

File: libs/pilight/protocols/433.92/logilink_switch2.c (hunk 3 above)
Comment (CurlyMoo) [reply libs/pilight/protocols/433.92/logilink_switch2.c]: > Did you mean the space behind createHigh(40, 45);

Yes

**Comment by hernae** [comment]
> That your commit icon doesn't show homer simpson.
I published the commit with the wrong email address (one without gravatar). I just linked that email to gravatar as well, dunno if its working. Otherwise I would have to rewrite the git history (never done that).


> Under the rewrite branch.
Okay, merging the changes into the rewrite branch will be no problem. I don't know how when I have time for the unit tests.

**Comment by CurlyMoo** [comment]
>  Otherwise I would have to rewrite the git history (never done that).

You always have to squash all commits into one before merging. A link can be found in the contribution guide.

File: libs/pilight/protocols/433.92/logilink_switch2.c [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+#include "../protocol.h"
+#include "../../core/binary.h"
+#include "../../core/gc.h"
+#include "logilink_switch2.h"
+
+#define PULSE_MULTIPLIER	3
+#define MIN_PULSE_LENGTH	279
+#define MAX_PULSE_LENGTH	289
+#define AVG_PULSE_LENGTH	284
+#define RAW_LENGTH				50
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/logilink_switch2.c]: Can you check the tabs here?

File: libs/pilight/protocols/433.92/logilink_switch2.c [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+	int unitcode = -1;
+	int state = -1; 
+	double itmp = 0;
+
+	if(json_find_number(code, "systemcode", &itmp) == 0)
+		systemcode = (int)round(itmp);
+	if(json_find_number(code, "unitcode", &itmp) == 0)
+		unitcode = (int)round(itmp);
+	if(json_find_number(code, "off", &itmp) == 0)
+		state=1;
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/logilink_switch2.c]: Can you add spaces between the equals sign?

File: libs/pilight/protocols/433.92/logilink_switch2.c [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+	protocol_set_id(logilink_switch2, "logilink_switch2");
+	protocol_device_add(logilink_switch2, "logilink_switch2", "Logilink Switches");
+	logilink_switch2->devtype = SWITCH;
+	logilink_switch2->hwtype = RF433;
+	logilink_switch2->minrawlen = RAW_LENGTH;
+	logilink_switch2->maxrawlen = RAW_LENGTH;
+	logilink_switch2->maxgaplen = MAX_PULSE_LENGTH*PULSE_DIV;
+	logilink_switch2->mingaplen = MIN_PULSE_LENGTH*PULSE_DIV;
+
+	options_add(&logilink_switch2->options, 's', "systemcode", OPTION_HAS_VALUE, DEVICES_ID, JSON_NUMBER, NULL, NULL);
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/logilink_switch2.c]: Can you add the regex here?

**Comment by CurlyMoo** [comment]
Can you also add a protocol page in the root docs folder following the `rst` syntax?

**Pull request closed without merging**